### ISSUE TITLE
Playback speed too many options

### ISSUE THREAD

**Issue by bluesxblues** [yuliskov/SmartTube#1645]
Playback speed too many options 1.00 1.05 1.10 1.2 1.3 1.5 ...

Please reduce options 1.00 1.25 1.5 1.75 2.00

Thank you

**Comment by stevensmedia** [comment]
Disagree. 1.4x is very useful for me on a lot of videos I watch. It's just right.  I'm sure for others, other speeds are also useful.

**Comment by costinmx** [comment]
I agree somehow there are too many speed options. IMHO, 0.1x step should do it (until 1.5x)
But on high bitrate videos like 4K/60FPS, sometimes you have to fine tune the speed in very precise small steps to match the amonunt of how much bandwith and CPU draw your device supports. 
Also, on devices with Motion Compensation technology, you may consider that some precise speed steps work better on related framerates (multiplying or dividing original video FPS with adjusted speed)

**Comment by zamazan4ik** [comment]
I suggest leave some "default" options and provide a possibility to enter own speed for special cases. Also, will be awesome if will be available an option to save own speed to the predefined set for resusing it later.

**Comment by Jamie-Landeg-Jones** [comment]
No, I use speed options between the ones you suggest. 1.15 is often optimum.

Why remove them, anyway? If you don't like one of the speed options, ignore it. It's hardly a hassle.

**Comment by cosimochellini** [comment]
Maybe give the option to the user to choose which speed options need to be displayed in the settings? 🤔, I think could be the best compromise

**Comment by san-kumar** [comment]
I feel the suggestion provided by @zamazan4ik is the best. A way to enter custom values and maybe have it added to the list would be just awesome. 

Please let me know if I can work on this feature and submit a pull request.

**Comment by hiroo916** [comment]
I'm not quite sure when the change happened since there have been a few updates recently (as of 2/23/23) but as of version 17.20, the speed options have been reduced to only 0.25 increments.

I'm here to request that more speed options be put back because the 0.25 jumps are too big.  Going from 1.0 to 1.25 is just too much on some videos.  For videos that I want to watch "naturally" but reduce the time, more granular speeds between 1.0 and 1.5 are needed.  Above 1.5, I think the 0.25 increment is ok.  

And I found the previous available 0.95 option really useful for watching foreign language videos that I am learning or have some comprehension of, because it made it easier to decode the words without making it sound unnaturally slow.  Below that 0.75, 0.5, 0.25 is ok.

Now for those who thought there were too many options, I also was once annoyed that there were so many options, because clicking and scrolling through the menu was a bit troublesome, especially in the previous UI where clicking the Clock icon would pop up the full menu of speeds and if you were at 3.0x and wanted to go back to 1.0x, you would have to scroll all the way back down.  However, I'd made the suggestion (not claiming sole credit) to have single click turn on/off speed-adjusted playback, and click-hold pop up the speed adjustment menu.  That reduced a lot of cases of having to click through the whole menu because you could go back and forth between 1.0 and selected speed with one toggle click. 

I later got the Rii K25 remote control for my Shield (because I was sick of changing batteries on the slim remote) and discovered a great feature: the PageUp/PageDown buttons worked on SmartTubeNext to adjust the playback speed up and down without going into any menus.  So this might be another option for people annoyed with the longer menu, find an alternative remote with PgUp/PgDown keys. Not sure what other ones work for this but the K25 does.

So please put my vote in for restoring the previous longer list of speed options: 0.25, 0.5, 0.75, 0.95, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.75, 2.0, 2.25, 2.5, 2.75, 3.0.

If it needs to be configurable in the Settings, maybe just have a option list of speeds from 0.25 to 3.0 in 0.05 increments and the user can select the ones they want to appear in the menu.

**Comment by costinmx** [comment]
You should really read the description before updating. From 17.20 there is a specific STN setting in place for setting this increments to smaller steps. 

Somebody should close these redundant topics

**Comment by hiroo916** [comment]
I looked through all the settings for 10 minutes for some type of option to adjust this before coming here. I couldn't find a related setting. Can you point out where it is?

**Comment by MailYouLater** [comment]
The option added in 17.20 is a toggle between the new 0.25x increments list of speeds, and the old list.

Video player > Misc > Long speed list

![Screenshots](https://user-images.githubusercontent.com/1279475/221297286-4a2b319f-809d-430f-bc18-dbd6686d6a9d.jpg)
